### DISCUSSION TITLE
Failed test cases from Thread Harness

### DISCUSSION THREAD

**Discussion by testter2** [openthread/openthread#7618] [Q&A] [answered]
Hello,
I'm using Thread Test Harness software Release 2.0.0 and I'm integrating platform with openthread cli example downloaded from the repo on 17/4/2022 and the test cases in following report are failing when I run it.
my setup consists of :-
DUT: ARC with firmware version OPENTHREAD/20191113; NONE; Apr 20 2022 14:23:36
3 NXP devices with firmware version: KTSTB_190306UC0

here is the report produced from running test cases :
[OpenThread_Commissioner_Consolidate Report_R2.0.0_2022_04_21-00_30_53.pdf](https://github.com/openthread/openthread/files/8525860/OpenThread_Commissioner_Consolidate.Report_R2.0.0_2022_04_21-00_30_53.pdf)

and this the sniffer captures :
[Captures.zip](https://github.com/openthread/openthread/files/8525871/Captures.zip)

I am using the following configurations:
#define OPENTHREAD_CONFIG_COMMISSIONER_ENABLE 1
#define OPENTHREAD_CONFIG_JOINER_ENABLE 1
#define OPENTHREAD_CONFIG_COAP_API_ENABLE 1
#define OPENTHREAD_CONFIG_MAC_FILTER_ENABLE 1
#define OPENTHREAD_CONFIG_REFERENCE_DEVICE_ENABLE 1
#define OPENTHREAD_CONFIG_IP6_SLAAC_ENABLE 1
#define OPENTHREAD_CONFIG_DHCP6_CLIENT_ENABLE 1

I need to know if there's a common reason for these failures or there's other configurations should be configured?

Thanks,

**Comment by jwhui** [comment, marked as answer]
Glancing at the pcap files, it appears that the DTLS handshake is occurring, but something in the exchange fails after the Change Cipher Spec method. This could be due to a mismatch in the PSKd between the Joiner and Commissioner.

By ARC firmware, are you referring to the [Synopsys platform](https://openthread.io/vendors/synopsys)? I don't have any experience with this platform. Have you tried reaching out directly to the platform vendor for support?

**Reply by testter2** [reply]
Thanks for your reply.
I think I was using wrong version for mbed-tls library, after changing it now test cases failed because of the following error:
Check if Commissioner sends encrypted JOIN_ENT.ntf to Joiner: : No 

here is the report produced from running test cases :
[OpenThread_Commissioner_Consolidate Report_R2.0.0_2022_04_22-02_29_46.pdf](https://github.com/openthread/openthread/files/8535876/OpenThread_Commissioner_Consolidate.Report_R2.0.0_2022_04_22-02_29_46.pdf)

if you have a reason for the previous failure, it would be very helpful.
